src/Routes/RosPage/RosPage.js

```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
 
     constructor(props) {
         super(props);
+
+        this.state = {
+            page: 1,
+            perPage: 5
+        };
+
+        this.onSetPage = this.onSetPage.bind(this);
+        this.onPerPageSelect = this.onPerPageSelect.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        this.props.fetchSystems();
+        this.fetchWithParams();
```

Comment:
I would call directly the dispatched fetch function with state. `fetchSystems({ page: this.state.page, perPage: this.state.perPage })`